crates/interfaces/src/executor.rs

```diff
@@ -13,7 +13,12 @@ pub trait BlockExecutor<T> {
     /// provided block's transactions internally. We use this to allow for calculating senders in
     /// parallel in e.g. staged sync, so that execution can happen without paying for sender
     /// recovery costs.
-    fn execute(&mut self, block: &Block, senders: Option<Vec<Address>>) -> Result<T, Error>;
+    fn execute(
+        &mut self,
+        block: &Block,
+        total_difficulty: U256,
```

Comment:
I really don't like how we're polluting the interface with both the `senders` and the `total_difficulty` option.. I think fine for now but e.g. if I was consuming this client-side, I'd make it so that `BlockExecutor` takes a block where transactions + senders are in 1 struct, and the total_difficulty param is not needed as the executor has some stat var over the block number to check instead of TD

Reply:
We used to use a type for tx with recovered senders but it was removed from all interfaces in a PR for some reason